File: 6toCiclo/DAMII/javaphp/app/src/main/java/com/example/javaphp/MainActivity.java
package com.example.javaphp;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.toolbox.JsonObjectRequest;
import com.android.volley.toolbox.Volley;
import org.json.JSONArray;
import org.json.JSONObject;
import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {
    RecyclerView recyclerView;
    ClienteAdapter adapter;
    List<Cliente> clientes;
    RequestQueue requestQueue;
    Button btnAgregarCliente;

    private static final String URL_LISTAR = Config.getServerURL() + "listar.php";


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        recyclerView = findViewById(R.id.recyclerView);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        clientes = new ArrayList<>();
        adapter = new ClienteAdapter(this, clientes);
        recyclerView.setAdapter(adapter);
        requestQueue = Volley.newRequestQueue(this);

        btnAgregarCliente = findViewById(R.id.btnAgregarCliente);
        btnAgregarCliente.setOnClickListener(v -> {
            Intent intent = new Intent(MainActivity.this, AgregarActivity.class);
            startActivity(intent);
        });

        listarClientes();
    }

    @Override
    protected void onResume() {
        super.onResume();
        listarClientes();
    }

    private void listarClientes() {
        JsonObjectRequest request = new JsonObjectRequest(
                Request.Method.GET, URL_LISTAR, null,
                response -> {
                    clientes.clear();
                    try {
                        JSONArray array = response.getJSONArray("clientes");
                        for (int i = 0; i < array.length(); i++) {
                            JSONObject clienteJson = array.getJSONObject(i);
                            Cliente cliente = new Cliente(
                                    clienteJson.getString("dni"),
                                    clienteJson.getString("nombre"),
                                    clienteJson.getString("correo"),
                                    clienteJson.getString("telefono"),
                                    clienteJson.getString("fecha_nacimiento"),
                                    clienteJson.getString("latitud"),
                                    clienteJson.getString("longitud")
                            );
                            clientes.add(cliente);
                        }
                        adapter.notifyDataSetChanged();
                    } catch (Exception e) {
                        e.printStackTrace();
                    }
                },
                Throwable::printStackTrace
        );

        requestQueue.add(request);
    }
}

File: 6toCiclo/DAMII/pagoempleados/app/src/main/java/pe/payka/pagoempleados/MainActivity.kt
package pe.payka.pagoempleados

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val etEmpleado = findViewById<EditText>(R.id.etEmpleado)
        val etHoras = findViewById<EditText>(R.id.etHoras)
        val etTarifa = findViewById<EditText>(R.id.etTarifa)
        val btnProcesar = findViewById<Button>(R.id.btnProcesar)
        val btnLimpiar = findViewById<Button>(R.id.btnLimpiar)

        val tvEmpleadoIngresado = findViewById<TextView>(R.id.tvEmpleadoIngresado)
        val tvTotalHoras = findViewById<TextView>(R.id.tvTotalHoras)
        val tvTotalTarifa = findViewById<TextView>(R.id.tvTotalTarifa)
        val tvTotalSueldoBruto = findViewById<TextView>(R.id.tvTotalSueldoBruto)
        val tvTotalEssalud = findViewById<TextView>(R.id.tvTotalEssalud)
        val tvTotalAFP = findViewById<TextView>(R.id.tvTotalAFP)
        val tvTotalSueldoNeto = findViewById<TextView>(R.id.tvTotalSueldoNeto)

        btnProcesar.setOnClickListener {
            val nombre = etEmpleado.text.toString()
            val horas = etHoras.text.toString().toDoubleOrNull() ?: 0.0
            val tarifa = etTarifa.text.toString().toDoubleOrNull() ?: 0.0

            val sueldoBruto = horas * tarifa
            val descuentoEssalud = sueldoBruto * 0.12
            val descuentoAFP = sueldoBruto * 0.10
            val sueldoNeto = sueldoBruto - descuentoEssalud - descuentoAFP

            tvEmpleadoIngresado.text = nombre
            tvTotalHoras.text = "%.2f".format(horas)
            tvTotalTarifa.text = "%.2f".format(tarifa)
            tvTotalSueldoBruto.text = "%.2f".format(sueldoBruto)
            tvTotalEssalud.text = "%.2f".format(descuentoEssalud)
            tvTotalAFP.text = "%.2f".format(descuentoAFP)
            tvTotalSueldoNeto.text = "%.2f".format(sueldoNeto)
        }

        btnLimpiar.setOnClickListener {
            etEmpleado.text.clear()
            etHoras.text.clear()
            etTarifa.text.clear()
            tvEmpleadoIngresado.text = ""
            tvTotalHoras.text = ""
            tvTotalTarifa.text = ""
            tvTotalSueldoBruto.text = ""
            tvTotalEssalud.text = ""
            tvTotalAFP.text = ""
            tvTotalSueldoNeto.text = ""
        }
    }
}
